main.tf

```
@@ -422,69 +422,90 @@ resource "zenml_service_connector" "gcp" {
 }
 
 resource "zenml_stack_component" "orchestrator" {
-  name      = "${var.zenml_stack_name == "" ? "terraform-${var.orchestrator}-${random_id.resource_name_suffix.hex}" : "${var.zenml_stack_name}-${var.orchestrator}"}"
-  type      = "orchestrator"
-  flavor    = var.orchestrator == "skypilot" ? "vm_gcp" : var.orchestrator
+  name   = var.zenml_stack_name == "" ? "terraform-${var.orchestrator}-${random_id.resource_name_suffix.hex}" : "${var.zenml_stack_name}-${var.orchestrator}"
+  type   = "orchestrator"
+  flavor = var.orchestrator == "skypilot" ? "vm_gcp" : var.orchestrator
 
   configuration = local.orchestrator_config[var.orchestrator]
 
   connector_id = contains(["local", "airflow"], var.orchestrator) ? "" : zenml_service_connector.gcp.id
 
   labels = {
-    "zenml:provider" = "gcp"
+    "zenml:provider"   = "gcp"
     "zenml:deployment" = "${var.zenml_stack_deployment}"
   }
 }
 
 # Step Operator
 resource "zenml_stack_component" "step_operator" {
-  name      = "${var.zenml_stack_name == "" ? "terraform-vertex-${random_id.resource_name_suffix.hex}" : "${var.zenml_stack_name}-vertex"}"
-  type      = "step_operator"
-  flavor    = "vertex"
+  name   = var.zenml_stack_name == "" ? "terraform-vertex-${random_id.resource_name_suffix.hex}" : "${var.zenml_stack_name}-vertex"
+  type   = "step_operator"
+  flavor = "vertex"
 
   configuration = {
-    region = "${data.google_client_config.current.region}",
+    region          = "${data.google_client_config.current.region}",
     service_account = "${google_service_account.zenml_sa.email}"
   }
 
   connector_id = zenml_service_connector.gcp.id
 
   labels = {
-    "zenml:provider" = "gcp"
+    "zenml:provider"   = "gcp"
     "zenml:deployment" = "${var.zenml_stack_deployment}"
   }
 }
 
 # Image Builder
 resource "zenml_stack_component" "image_builder" {
-  name      = "${var.zenml_stack_name == "" ? "terraform-gcp-${random_id.resource_name_suffix.hex}" : "${var.zenml_stack_name}-gcp"}"
-  type      = "image_builder"
-  flavor    = "gcp"
+  name   = var.zenml_stack_name == "" ? "terraform-gcp-${random_id.resource_name_suffix.hex}" : "${var.zenml_stack_name}-gcp"
+  type   = "image_builder"
+  flavor = "gcp"
 
   configuration = {}
 
   connector_id = zenml_service_connector.gcp.id
 
   labels = {
-    "zenml:provider" = "gcp"
+    "zenml:provider"   = "gcp"
+    "zenml:deployment" = "${var.zenml_stack_deployment}"
+  }
+}
+
+# Experiment Tracker
+resource "zenml_stack_component" "experiment_tracker" {
+  name   = var.zenml_stack_name == "" ? "terraform-gcp-${random_id.resource_name_suffix.hex}" : "${var.zenml_stack_name}-gcp"
+  type   = "experiment_tracker"
+  flavor = "vertex"
+
+  configuration = {
+    project        = "${data.google_client_config.current.project}",
+    location       = "${data.google_client_config.current.region}"
+    staging_bucket = "gs://${google_storage_bucket.artifact_store.name}"
+  }
+
+  connector_id = zenml_service_connector.gcs.id
```

You're referencing the GCS service connector here, but this is specialized for GCS resources. What you need for the Vertex experiment tracker is a generic GCP service connector, similar to the orchestrator or step operator.

```suggestion
  connector_id = zenml_service_connector.gcp.id
```